File: server/src/Models/DMToolkitModels/Instances/SkillInstance.cs
using DMToolkit.API.Models.DMToolkitModels.Definitions;

namespace DMToolkit.API.Models.DMToolkitModels.Instances;

public class SkillInstance : IDefinitionInstance<SkillDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public bool IsProficient { get; set; } = false;
    public string AbilityScoreInstanceId { get; set; } = string.Empty;
    public AbilityScoreInstance? AbilityScoreInstance { get; set; } = null;
    public string DefinitionId { get; set; } = string.Empty;
    public SkillDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/skillInstance.ts
import type { SkillDefinition } from "../definitions/skillDefinition";

export interface SkillInstance {
    id: string;
    isProficient: boolean;
    definition: SkillDefinition;
}

export interface SkillInstanceDto {
    id: string;
    isProficient: boolean;
    definitionId: string;
}